test/e2e/singlecluster/pod_test.go

```diff
@@ -552,4 +553,110 @@ var _ = ginkgo.Describe("Pod groups", func() {
+				util.MustCreate(ctx, k8sClient, defaultRF)
+			})
+
+			ginkgo.By("Creating two cluster queues in the same cohort", func() {
+				// cluster-queue-1 has 8 CPU nominal quota
+				clusterQueue1 = testing.MakeClusterQueue("cluster-queue-1").
+					Cohort("bug-test").
+					ResourceGroup(
+						*testing.MakeFlavorQuotas("default-flavor").
+							Resource(corev1.ResourceCPU, "8").
+							Obj(),
+					).
+					Obj()
+				util.MustCreate(ctx, k8sClient, clusterQueue1)
+
+				// cluster-queue-2 has 0 CPU nominal quota (will need to borrow)
+				clusterQueue2 = testing.MakeClusterQueue("cluster-queue-2").
+					Cohort("bug-test").
+					ResourceGroup(
+						*testing.MakeFlavorQuotas("default-flavor").
+							Resource(corev1.ResourceCPU, "0").
+							Obj(),
+					).
+					Obj()
+				util.MustCreate(ctx, k8sClient, clusterQueue2)
+			})
+
+			ginkgo.By("Creating local queues for the corresponding cluster queues", func() {
+				localQueue1 = testing.MakeLocalQueue("local-queue-1", ns.Name).
+					ClusterQueue("cluster-queue-1").Obj()
+				util.MustCreate(ctx, k8sClient, localQueue1)
+
+				localQueue2 = testing.MakeLocalQueue("local-queue-2", ns.Name).
+					ClusterQueue("cluster-queue-2").Obj()
+				util.MustCreate(ctx, k8sClient, localQueue2)
+			})
+
+			ginkgo.By("Verify both cluster queues are active", func() {
+				util.ExpectClusterQueuesToBeActive(ctx, k8sClient, clusterQueue1, clusterQueue2)
+			})
+
+			ginkgo.By("Delete cluster-queue-1 which has 8 CPU nominal quota", func() {
+				gomega.Expect(k8sClient.Delete(ctx, clusterQueue1)).To(gomega.Succeed())
+				util.ExpectObjectToBeDeleted(ctx, k8sClient, clusterQueue1, true)
+			})
+
+			ginkgo.By("Create a pod requesting 8 CPUs to cluster-queue-2", func() {
+				pod := &corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "pod-8-cpu",
+						Namespace: ns.Name,
+						Labels: map[string]string{
+							constants.QueueLabel: "local-queue-2",
+						},
+					},
+					Spec: corev1.PodSpec{
+						RestartPolicy: corev1.RestartPolicyNever,
+						Containers: []corev1.Container{
+							{
+								Name:    "sleep",
+								Image:   "busybox:stable",
+								Command: []string{"sleep", "infinity"},
+								Resources: corev1.ResourceRequirements{
+									Requests: corev1.ResourceList{
+										corev1.ResourceCPU: resource.MustParse("8"),
+									},
+								},
+							},
+						},
+					},
+				}
+				util.MustCreate(ctx, k8sClient, pod)
+
+				ginkgo.By("Verify the pod should remain scheduling gated (not admitted)", func() {
+					gomega.Consistently(func(g gomega.Gomega) {
+						var updatedPod corev1.Pod
+						g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(pod), &updatedPod)).To(gomega.Succeed())
+						g.Expect(updatedPod.Spec.SchedulingGates).To(gomega.ContainElement(corev1.PodSchedulingGate{
+							Name: podconstants.SchedulingGateName,
+						}))
+					}, util.ConsistentDuration, util.ShortInterval).Should(gomega.Succeed())
+				})
+			})
+		})
+	})
```

Comment:
Note for future me: You'll notice that this integration test does not include a check on whether the workload reservation was removed. 

See: https://github.com/kubernetes-sigs/kueue/issues/5937#issuecomment-3077597464

> Even though the quota for cohort is reduced, the Workload still has the QuotaReserved condition. Kueue does not have a mechanism to preempt due to reduced quota (either by lowering the numbers or deleting a CQ contributing to the cohort's quota). So, as a result CQs and cohorts may run over-committed for a while.